com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDSubTarget.cs

```diff
@@ -91,8 +92,9 @@ public override void Setup(ref TargetSetupContext context)
         {
             context.AddAssetDependency(kSourceCodeGuid, AssetCollection.Flags.SourceDependency);
             context.AddAssetDependency(subTargetAssetGuid, AssetCollection.Flags.SourceDependency);
+            var inspector = VFXSubTarget.IsConfigured() ? VFXHDRPSubTarget.Inspector : customInspector;
             if (!context.HasCustomEditorForRenderPipeline(typeof(HDRenderPipelineAsset)))
-                context.AddCustomEditorForRenderPipeline(customInspector, typeof(HDRenderPipelineAsset));
+                context.AddCustomEditorForRenderPipeline(inspector, typeof(HDRenderPipelineAsset));
```

Comment:
In case of VFX generation we override the inspector with the one suited for VFX Context UI.